CMakeLists.txt

```diff
@@ -17,21 +17,17 @@ find_package(pluginlib REQUIRED)
 find_package(tf2 REQUIRED)
 find_package(tf2_ros REQUIRED)
 find_package(tf2_geometry_msgs REQUIRED)
-find_package(Boost REQUIRED COMPONENTS system filesystem thread)
+find_package(Boost REQUIRED)
 find_package(OpenSSL REQUIRED)
 
-include_directories(
-  include
-  ${OPENSSL_INCLUDE_DIR}
-)
-
 set(warehouse_srcs
   src/database_loader.cpp
   src/transform_collection.cpp)
 
 add_library(warehouse_ros SHARED ${warehouse_srcs})
+target_link_libraries(warehouse_ros PUBLIC OpenSSL::Crypto)
 set_target_properties(warehouse_ros PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
-ament_target_dependencies(warehouse_ros
+ament_target_dependencies(warehouse_ros SYSTEM PUBLIC
```

Comment:
Any reason for marking these dependencies as `SYSTEM` .? see [this comment](https://github.com/ros-planning/geometric_shapes/pull/186#discussion_r623989443) on why it may cause some problems in certain cases

Reply:
Oh, I wasn't aware of that. I'll make a PR to document that in ament_catkin.